### ISSUE TITLE
Unable to connect when using ansible_ssh_pass

### ISSUE THREAD

**Issue by craph** [rundeck-plugins/ansible-plugin#235]
Hello,

I am unable to view my nodes in the rundeck interface. I have create as mention in the documentation an inventory.ini file like this :
```ini
[test-group]
local-thing      ansible_connection=local      ansible_python_interpreter=/usr/bin/python3
my-first-server  ansible_host=1.2.3.4          ansible_user=root    ansible_ssh_pass=SUPER_SECRET
```

But I have the following error in the log of rundeck :
` FAILED! => {"msg": "to use the 'ssh' connection type with passwords, you must install the sshpass program"`

but when I connect to the docker container I cannot install sshpass because I don't have the password for the sudo...

Can you help me ?

Best regards,

**Comment by craph** [comment]
The package sshpass is missing in the DockerFile. Could you add it please ?

Thank you very much.

**Comment by frozenice** [comment]
[Done](https://github.com/Batix/rundeck-ansible-plugin/commit/b90db80b64f95ca815eaad1197a50adcd0b9298f). Can you try with [3.0.1](https://github.com/Batix/rundeck-ansible-plugin/releases/tag/3.0.1)?

**Comment by zopar** [comment]
It is strange how the plugin set the ansible command.
If you select to use password and set an user, the command will use --ask-password and the ansible-playbook command hangs because rundeck do not answer to the prompt "SSH password:" generated by ansible.
If you omit the password inside rundek interface, the plugin completely ignore ssh user and password set in the hosts.ini I think because it pass the extra var file that is in tmp
Basically it is not possible to correctly setup a connection with user and password defined in the hosts.ini
We need to pass in any case an hosts.ini why you do not simplify things and launch only the ansible-playbook and keep users to manage autentication via standard hosts.ini vars and ansible.cfg?

**Comment by zopar** [comment]
Ok I confirm the issue:
you pass extra vars in your command, this take precedence over ansible_ssh_user and password set in hosts.ini. For example if you set gather fact this include the extra vars and does not works, if you remove gather facts, it works because you do not set extra vars and ansible uses user and password from hosts.ini .
If you set the password inside rundeck, the command use --ask-pass and this is incorrect because this prompt password and you do not want this because you are in "background" not in a terminal.

Possible fix:
if an user select to use, user and password, do not use --ask-pass but insert the user and password inside your extra vars file, so user do not needs to specify it on hosts.ini
But remember this limit the possibility to specify multiple user and password by ansible groups.

I think the better option is to disable this type of options from interface and simply use classical hosts.ini that is more flexible.

**Comment by craph** [comment]
Any news on this issue ?

**Comment by fdevans** [comment]
Sorry we missed this. If it's still a problem please open a new Issue with current Rundeck version details, and a deeper description of how this can be repeated.
